e2e/src/pages/app.po.ts

```diff
@@ -54,12 +56,26 @@ export class AppPage {
   async waitForElement(selector: Locator) {
     await browser.wait(async () => {
       return (await element(selector)).isPresent();
-    }, 10000, 'failed to load search results');
+    }, 30000, 'failed to load search results');
   }
 
   async waitForElementsArray(selector: Locator) {
     await browser.wait(async () => {
       return (await element.all(selector).isPresent());
-    }, 10000, 'failed to load search results');
+    }, 30000, 'failed to load search results');
+  }
+
+  async getItem(el: ElementFinder) {
+    const item = protractor.promise.defer();
+    browser.wait(function() {
+      return element(by.id('el')).getAttribute('attribute').then(function(value) {
```

Comment:
please don't use .then use await

Reply:
I tried different ways to get the value from getAttribute() method but some reason that's not working that's why i used old fashion of using promise

Reply:
Let's discuss this